synapse/handlers/room.py

```diff
@@ -1179,11 +1191,11 @@ async def filter_evts(events: List[EventBase]) -> List[EventBase]:
         # filter_evts can return a pruned event in case the user is allowed to see that
         # there's something there but not see the content, so use the event that's in
         # `filtered` rather than the event we retrieved from the datastore.
-        results["event"] = filtered[0]
+        event = filtered[0]
```

Comment:
Yeah, we are...I initially had converted that too and then stopped. I think it got messy, let me give it a try again though!

Reply:
I've updated this to not mutate those results, it got a bit messy though. Let me know if you'd prefer I back it out. See 177783111825eb78c76c797fe033dd0ee1b2311d.

Reply:
To be honest I don't mind too much either way. (I was mostly checking that we hadn't missed a bit of work that we'd intended to do, judging by the commit message).

I like that the change makes it a bit easier to see what you get out of `get_events_around` (i.e. a thing with specific fields rather than an arbitrary dictionary), but I don't really have a strong opinion here. Happy to defer to your judgement!